### PR TITLE
Solr synchronization improvements

### PR DESCRIPTION
**Pull request by fedys**
#287

### CODE REVIEW

File: module/Solr/src/Solr/Listener/JobEventSubscriber.php [hunk 1]
```diff
@@ -49,154 +73,101 @@ public function __construct(Manager $manager)
...
+        }
+            
+        // check if the job is active
+        if ($document->isActive()) {
+            // mark it for commit
+            $this->add[] = $document;
+        } else {
+            // mark it for delete
+            $this->delete[] = $document;
+        }
```
Comment (TiSiE) [inline module/Solr/src/Solr/Listener/JobEventSubscriber.php]: No, the job should **only** be removed from solr index, when the state changed to "inactive".

Reason: 
Job could be edited after publishing, and these edition must be reviewed, but the job WITHOUT modification should still be available in the jobboard.

File: module/Solr/src/Solr/Listener/JobEventSubscriber.php [hunk 2]
```diff
@@ -49,154 +73,101 @@ public function __construct(Manager $manager)
...
+            $document = $this->entityToDocumentFilter->filter($job);
+            $client->addDocument($document);
+        }
+        
+        // process jobs for delete
+        foreach ($this->delete as $job) {
+            $client->deleteByIds($this->entityToDocumentFilter->getDocumentIds($job));
         }
     }
+
```
Comment (TiSiE) [inline module/Solr/src/Solr/Listener/JobEventSubscriber.php]: Isn't postUpdate hook called after EACH document?

Maybe we do not need this hook and handle all of it in the postFlush hook?

**Comment by cbleek** [comment]
the job should *only* be removed from solr index, when the state changed 
to "inactive"*OR "expired"*

File: module/Solr/src/Solr/Listener/JobEventSubscriber.php (hunk 2 above)
Comment (fedys) [inline module/Solr/src/Solr/Listener/JobEventSubscriber.php]: @TiSiE you are right

**Pull request merged**